File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s=s.toLowerCase();
    let start=0;
    let end=s.length-1;
    while(start<end){
        if(!s[start].match(/[a-z0-9]/)){
            ++start;
        }
        else if(!s[end].match(/[a-z0-9]/)){
            --end;
        }
        else if(s[start]===s[end]){
                ++start;
                --end;
    
            }
            else{
                return false;
            }
        
    }
    return true;
    
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
           StringBuilder filtered = new StringBuilder();
        StringBuilder rev = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                char lower = Character.toLowerCase(c);
                filtered.append(lower);
                rev.insert(0, lower);
            }
        }
        return filtered.toString().equals(rev.toString());
        
    }
}